Scripts/LaunchSidechainMasternode.ps1

```diff
@@ -0,0 +1,484 @@
+    $LocalHeight
+}
+
+function Get-LocalIndexerHeight 
+{
+    $IndexStatsRequest = Invoke-WebRequest -Uri http://localhost:$API/api/BlockStore/addressindexertip
+    $IndexStats = ConvertFrom-Json $IndexStatsRequest
+    $LocalIndexHeight = $IndexStats.tipHeight
+    $LocalIndexHeight
+}
+
+function Get-BlockStoreStatus
+{
+    $FeatureStatus = Invoke-WebRequest -Uri http://localhost:$API/api/Node/status | ConvertFrom-Json | Select-Object -ExpandProperty featuresData
+    $BlockStoreStatus = $FeatureStatus | Where-Object { $_.namespace -eq "Stratis.Bitcoin.Features.BlockStore.BlockStoreFeature" }
+    $BlockStoreStatus.state
+}
+
+function Shutdown-Dashboard
+{
+    Write-Host "Shutting down Stratis Masternode Dashboard..." -ForegroundColor Yellow
+    Start-Job -ScriptBlock { Invoke-WebRequest -Uri http://localhost:37000/shutdown } | Out-Null
+        While ( Test-Connection -TargetName 127.0.0.1 -TCPPort 37000 -ErrorAction SilentlyContinue )
+        {
+            Write-Host "Waiting for Stratis Masternode Dashboard to shut down" -ForegroundColor Yellow
+            Start-Sleep 5
+        }
+}
+
+function Check-TimeDifference
+{
+    Write-Host "Checking UTC Time Difference" -ForegroundColor Cyan
+    $timeDif = New-TimeSpan -Start (Get-Date).ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ") -End ( Invoke-WebRequest http://worldtimeapi.org/api/timezone/UTC | ConvertFrom-Json | Select-Object -ExpandProperty utc_datetime ) | Select-Object -ExpandProperty TotalSeconds
+    if ( $timeDif -gt 8 )
+    {
+        Write-Host "ERROR: System Time is not accurate. Currently $timeDif seconds diffence with actual time! Correct Time & Date and restart" -ForegroundColor Red
+        Start-Sleep 30
+        Exit
+    }
+        Else
+        {
+            Write-Host "SUCCESS: Time difference is $timeDif seconds" -ForegroundColor Green
+            Write-Host ""
+        }
+}
+
+#Create DataDir(s)
+if ( -not ( Get-Item -Path $mainChainDataDir -ErrorAction SilentlyContinue ) )
+{
+    New-Item -ItemType Directory -Path $mainChainDataDir
+}
+
+if ( -not ( Get-Item -Path $sideChainDataDir -ErrorAction SilentlyContinue ) )
+{
+    New-Item -ItemType Directory -Path $sideChainDataDir
+}
+
+#Gather Federation Detail
+if ( -not ( Test-Path $sideChainDataDir\federationKey.dat ) ) 
+{
+    $miningDAT = Read-Host "Please Enter the full path to the federationKey.dat"
+    Copy-Item $miningDAT -Destination $sideChainDataDir -Force -ErrorAction Stop
+}
+
+#Establish Node Type
+if ( $multiSigMnemonic -ne $null )
+{
+    $NodeType = "50K"
+}
+
+#Check for Completion
+if ( $NodeType -eq "50K" )
+{
+    if ( -not ( $multiSigPassword ) ) { $varError = 1 }
+}
+    Else
+    {
+        if ( -not ( $miningPassword ) ) { $varError = 1 }
+    }
+
+if ( -not ( $mainChainDataDir )  ) { $varError = 1 }
+if ( -not ( $sideChainDataDir )  ) { $varError = 1 }
+if ( -not ( Test-Path $sideChainDataDir/federationKey.dat ) ) { $varError = 1 }
+
+if ( $varError -eq '1' )  
```

Comment:
Inconsistent use of data-types. Setting int by comparing against a string.
I think that should be a boolean

Set variable value before its first use `$varError = $false`
Set it to flag the error `$varError = $true`
Assert: `if ( $varError )`

Reply:
Done